api/process.go

```diff
@@ -0,0 +1,435 @@
+package api
+
+// Implements the task service functions for processes
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/cedana/cedana/api/services/task"
+	"github.com/rs/xid"
+	"github.com/shirou/gopsutil/v3/process"
+	"github.com/spf13/viper"
+	"go.opentelemetry.io/otel/attribute"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	OUTPUT_FILE_PATH  string      = "/var/log/cedana-output.log"
+	OUTPUT_FILE_PERMS os.FileMode = 0o777
+	OUTPUT_FILE_FLAGS int         = os.O_WRONLY | os.O_CREATE | os.O_TRUNC
+)
+
+var DB_BUCKET_JOBS = []byte("jobs")
+
+func (s *service) Start(ctx context.Context, args *task.StartArgs) (*task.StartResp, error) {
+	if args.Task == "" {
+		args.Task = viper.GetString("client.task")
+	}
+
+	state := &task.ProcessState{}
+	pid, err := s.run(ctx, args)
+	if err != nil {
+		// TODO BS: this should be at market level
+		s.logger.Error().Err(err).Msgf("failed to run task, attempt %d", 1)
+		return nil, status.Error(codes.Internal, "failed to run task")
+		// TODO BS: replace doom loop with just retrying from market
+	}
+
+	state.JobState = task.JobState_JOB_RUNNING
+	state.PID = pid
+	if args.JID == "" {
+		state.JID = xid.New().String()
+	} else {
+		existingState, err := s.getState(args.JID)
+		if err != nil {
+			return nil, status.Error(codes.NotFound, "failed to check existing jobs")
+		}
```

Comment:
This fails immediately when starting a process w/ exec in a clean setup (no /tmp/cedana.db file) with:
`2:18PM ERR start task failed error="rpc error: code = NotFound desc = failed to check existing jobs" file=/home/nravic/go/src/github.com/cedana/cedana/cmd/exec.go line=76`


My theory is nothing is being persisted to the db after the run (which is successful fwiw, I see a pid being spawned).

Reply:
On it.

Reply:
Fixed. So any get call on a fresh DB was failing. Because put commands were the only ones doing the necessary bucket creation etc.